var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -61,14 +61,24 @@ def mpi_dependent_options(self):
                 errors = ['incompatible variants given'] + errors
                 raise RuntimeError('\n'.join(errors))
         else:
-            compiler_opts = [
-                '--with-mpi=1',
-                '--with-mpi-dir=%s' % self.spec['mpi'].prefix,
-            ]
+            if self.compiler.name == "clang":
+                compiler_opts = [
+                    '--with-mpi=1',
+                    '--with-cc=%s  -Qunused-arguments' % join_path(self.spec['mpi'].prefix.bin, 'mpicc'), # Avoid confusing PETSc config by clang: warning: argument unused during compilation
```

Comment:
@eschnett they are not used in PETSc configure system. It does not work like that there, believe me. See http://lists.mcs.anl.gov/pipermail/petsc-users/2016-March/028910.html